homeassistant/components/tesla/__init__.py

```diff
@@ -186,11 +186,7 @@ def __init__(self, tesla_device, controller, config_entry):
         self._name = self.tesla_device.name
         self.tesla_id = slugify(self.tesla_device.uniq_name)
         self._attributes = {}
-        self._icon = (
-            SENSOR_ICONS[self.tesla_device.type]
-            if self.tesla_device.type and self.tesla_device.type in SENSOR_ICONS.keys()
-            else None
-        )
+        self._icon = ICONS.get(self.tesla_device.type, None)
```

Comment:
`self._icon = ICONS.get(self.tesla_device.type)`

dict.get() will return `None` by default if the key isn’t found.

https://www.tutorialspoint.com/python/dictionary_get.htm